File: Introduction to Data Structures: Hash Table/3. Practical Application - Hash Map/Contains Duplicate II/ContainsDuplicateIIHashMap.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {

        Map<Integer, Integer> hashMap = new HashMap<>(); 

        for (int i = 0; i < nums.length; i++) {

            if ( hashMap.containsKey(nums[i]) && i - hashMap.get(nums[i]) <= k ) {

                return true;
            }

            hashMap.put(nums[i], i);
        }
        
        return false;
    }
}

File: Introduction to Data Structures: Hash Table/3. Practical Application - Hash Map/Contains Duplicate II/ContainsDuplicateIIHashMap.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:

        hashMap = {}

        for i in range(0, len(nums)):

            if hashMap.get(nums[i]) != None:

                # print(i, hashMap.get(nums[i]), k)

                if i - hashMap.get(nums[i]) <= k:

                    return True

            hashMap[nums[i]] = i

        return False
